include/world_builder/topography/interface.h

```diff
@@ -0,0 +1,144 @@
+/*
+  Copyright (C) 2018-2024 by the authors of the World Builder code.
+
+  This file is part of the World Builder.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef WORLD_BUILDER_TOPOGRAPHY_MODELS_INTERFACE_H
+#define WORLD_BUILDER_TOPOGRAPHY_MODELS_INTERFACE_H
+
+#include "world_builder/topography/interface.h"
+
+#include "world_builder/parameters.h"
+#include "world_builder/types/string.h"
+
+
+namespace WorldBuilder
+{
+  class World;
+
+  namespace TopographyModel
+  {
+
+    class ObjectFactory;
+
+    /**
+     * This class is an interface for the specific coordinate systems.
+     */
+    class Interface
+    {
+      public:
+        /**
+         * constructor
+         */
+        Interface();
```

Comment:
since the constructor is the default constructor anyway you can remove this declaration.